indykite/resource_event_sink.go

```diff
@@ -184,33 +240,78 @@ func resourceEventSinkFlatten(
 ) diag.Diagnostics {
 	var d diag.Diagnostics
 	eventSink := resp.GetConfigNode().GetEventSinkConfig()
-	var results []map[string]any //nolint:prealloc // no prealloc
+	var results []map[string]any
 	for key, p := range eventSink.GetProviders() {
-		var oldPassword any
-		if val, ok := data.Get(providersKey).([]any); ok && len(val) > 0 {
-			kafkaMap, _ := val[0].(map[string]any)[kafkaKey].([]any)[0].(map[string]any)
-			oldPassword = kafkaMap[passwordKey]
-		}
-		kafkaConfig := map[string]any{
-			brokersKey:  p.GetKafka().GetBrokers(),
-			topicKey:    p.GetKafka().GetTopic(),
-			usernameKey: p.GetKafka().GetUsername(),
-			// Password is not retrieved from response, but omitting here would result in removing.
-			// First read old value and set it here too.
-			passwordKey: oldPassword,
-		}
+		switch p.GetProvider().(type) {
+		case *configpb.EventSinkConfig_Provider_Kafka:
+			var oldPassword any
+			if val, ok := data.Get(providersKey).([]any); ok && len(val) > 0 {
+				for _, y := range val {
+					if kafkaList, ok := y.(map[string]any)[kafkaKey].([]any); ok && len(kafkaList) > 0 {
+						kafka := kafkaList[0].(map[string]any)
+						oldPassword = kafka[passwordKey]
+					}
+				}
+			}
+			kafkaConfig := map[string]any{
+				brokersKey:  p.GetKafka().GetBrokers(),
+				topicKey:    p.GetKafka().GetTopic(),
+				usernameKey: p.GetKafka().GetUsername(),
+				// Password is not retrieved from response, but omitting here would result in removing.
+				// First read old value and set it here too.
+				passwordKey: oldPassword,
+			}
 
-		if p.GetKafka().GetDisableTls() {
-			kafkaConfig[disableTLSKey] = p.GetKafka().GetDisableTls()
-		}
-		if p.GetKafka().GetTlsSkipVerify() {
-			kafkaConfig[tlsSkipVerifyKey] = p.GetKafka().GetTlsSkipVerify()
-		}
-		result := map[string]any{
-			providerNameKey: key,
-			kafkaKey:        []any{kafkaConfig},
+			if p.GetKafka().GetDisableTls() {
+				kafkaConfig[disableTLSKey] = p.GetKafka().GetDisableTls()
+			}
+			if p.GetKafka().GetTlsSkipVerify() {
+				kafkaConfig[tlsSkipVerifyKey] = p.GetKafka().GetTlsSkipVerify()
+			}
```

Comment:
Style: The `if` is not needed here at all. You can just do this:

```diff
			kafkaConfig := map[string]any{
				brokersKey:  p.GetKafka().GetBrokers(),
				topicKey:    p.GetKafka().GetTopic(),
				usernameKey: p.GetKafka().GetUsername(),
				// Password is not retrieved from response, but omitting here would result in removing.
				// First read old value and set it here too.
				passwordKey: oldPassword,
+				disableTLSKey: p.GetKafka().GetDisableTls(),
+				tlsSkipVerifyKey: p.GetKafka().GetTlsSkipVerify(),
			}
```

Reply:
updated